linked_list.rb

```diff
@@ -132,19 +128,73 @@ def visit
 
   # method to delete the first node found with specified value
   def delete(value)
-    puts "Not implemented"
+    current = @head
+    prevNode = nil
+    if current.data == value
+      @head = current.next
+      return
+    end
+    while current.data != value
+      prevNode = current
+      current.next ? current = current.next : break
+    end
+
+    if current.data == value
+      if current.next
+        prevNode.next = current.next
+      else
+        prevNode.next = nil
+      end

```

Comment:
Optimization: you could replace lines 143 through 147 with just line 144, i.e. `prevNode.next = current.next`. If current.next is nil, line 144 will just set prevNode.next to nil.